buildsystem/coverage_report.gradle

```diff
@@ -13,15 +13,63 @@ jacoco {
     version = versions.jacoco
 }
 
-def variants = null
-if (android.hasProperty('libraryVariants')) {
-    variants = android.libraryVariants
-} else if (android.hasProperty('applicationVariants')) {
-    variants = android.applicationVariants
+afterEvaluate { project ->
+    def ignoreList = jacocoIgnoreList
+    def projectName = project.name
+    if (ignoreList.contains(projectName)) {
+        println "Jacoco: ignoring project ${projectName}"
+        return false
+    }
+    def threshold = project.hasProperty('jacocoCoverageThreshold')
+            ? project.jacocoCoverageThreshold
+            : project.jacocoCoverageThresholdDefault
+    setupTestExistenceValidationTask()
+    if (isAndroidModule(project)) {
+        setupAndroidReporting()
+        setupAndroidCoverageVerification(threshold)
+    } else {
+        setupKotlinReporting()
+        setupKotlinCoverageVerification(threshold)
+    }
+}
+
+def setupKotlinReporting() {
+    jacocoTestReport {
+        dependsOn test
+        dependsOn createDebugCoverageReport
+        reports {
+            csv.enabled false
+            xml.enabled true
+            html {
+                enabled true
+                destination file("${buildDir}/coverage-report")
+            }
+        }
+    }
 }
-if (variants != null) {
-    variants.all { variant ->
-        def variantName = variant.name.capitalize()
+
+def setupAndroidReporting() {
+    tasks.withType(Test) {
+        // Whether or not classes without source location should be instrumented
+        jacoco.includeNoLocationClasses true
+    }
+
+    task jacocoTestReport(
+            type: JacocoReport,
+            dependsOn: [
+                    'testDebugUnitTest',
+                    'createDebugCoverageReport'
+            ]
+    ) {
+        reports {
+            csv.enabled false
+            xml.enabled true
+            html {
+                enabled true
+                destination file("${buildDir}/coverage-report")
+            }
+        }
+        // Change as needed
         def fileFilter = [
```

Comment:
Maybe it is also worth including `**/*Binding*.*` ?

Reply:
I'd exclude it for the threshold check, but not for the coverage list.

Reply:
Maybe we should exclude all files in `intermediate` and `generated`?

Reply:
Unfortunately, we cannot exclude by directory. The files are taken from the compiled classes directory ($buildDir/tmp/kotlin-classes/debug). We can only exclude based on package or file name patterns :-(